fieldservice/models/fsm_equipment.py

```diff
@@ -45,3 +51,36 @@ def _onchange_branch_id(self):
     @api.onchange('district_id')
     def _onchange_district_id(self):
         self.region_id = self.district_id.region_id
+
+    @api.model
+    def _read_group_stage_ids(self, stages, domain, order):
+        stage_ids = self.env['fsm.stage'].search([('stage_type',
+                                                   '=', 'equipment')])
+        return stage_ids
+
+    def _default_stage_id(self):
+        return self.env['fsm.stage'].search([('stage_type', '=', 'equipment'),
+                                             ('sequence', '=', '1')])
+
+    def advance_stage(self):
+        seq = self.stage_id.sequence
+        next_stage = self.env['fsm.stage'].search(
+            [('stage_type', '=', 'equipment'), ('sequence', '=', seq+1)])
+        self.stage_id = next_stage
+        self._onchange_stage_id()
+
+    @api.onchange('stage_id')
```

Comment:
@osi-scampbell : you can check/compare the highest sequence stage name with following optimal source:
def _onchange_stage_id(self):
    heighest_stage = self.env['fsm.stage'].search(
        [('stage_type', '=', 'equipment')],
        order='sequence desc',
        limit=1
    )
    if self.stage_id.name ==heighest_stage.name:
            self.hide = True
        else:
            self.hide = False